src/sockets/sockets.ts

```diff
@@ -2782,8 +2782,8 @@ function newRoom(dataObj) {
 function joinRoom(roomId, inputPassword) {
   // console.log("inputpassword: " + inputPassword);
 
-  // if the room exists
-  if (rooms[roomId]) {
+  // if the room exists and the player is not currently in a room
+  if (rooms[roomId] && this.request.user.inRoomId === undefined) {
```

Comment:
Would prefer `!this.request.user.inRoomId` :). 
Helps us catch the case where it's 0, or null as well.